solr/core/src/java/org/apache/solr/handler/loader/XMLLoader.java

```diff
@@ -429,7 +434,18 @@ public SolrInputDocument readDoc(XMLStreamReader parser) throws XMLStreamExcepti
             break;
           } else if ("field".equals(parser.getLocalName())) {
             // should I warn in some text has been found too
-            Object v = isNull ? null : text.toString();
+            Object v;
```

Comment:
The interesting part is below, just using old style if then else for better readability, and yes it's required for binary support when using XML. The corresponding AtomicUpdateTest will fail otherwise

Reply:
So, just to make sure I understand: what is "binary XML" here?  Is this XML where some node in the XML doc has binary content?  Is this normal XML where the whole XML doc/string has been encoded using some binary format for compression or quicker transmission?

> this was the reason binary XML support was not working at least since 6.6.2
Interesting.  Is this something that Solr claimed to support or had support for at some point?  Or this is a new ability that Solr has never had that you're adding here?

Reply:
> So, just to make sure I understand: what is "binary XML" here? Is this XML where some node in the XML doc has binary content? Is this normal XML where the whole XML doc/string has been encoded using some binary format for compression or quicker transmission?
> 
There are several ways to transport binary data over XML, i used the one which seems to be the clearest most readable one from my point of view, which is also the proposed way on xml.com: https://www.xml.com/pub/a/98/07/binary/binary.html

> > this was the reason binary XML support was not working at least since 6.6.2
> 
> Interesting. Is this something that Solr claimed to support or had support for at some point? Or this is a new ability that Solr has never had that you're adding here?

Some versions before 6.6.2 EmbeddedSolrServer used XMLCodec as default codec. When people start using it more in there test cases there where some issues regarding enums, etc.. so the codec was switched to JavaBinCodec which was the default when using SolrJ anyway. So this feature was not needed any more, anyway as far as i know all the other codecs are supporting transport of binary data. So if anyone want to use XML when talking to Solr he can send binaries now, it is also required for the new tests, the would fail otherwise.

Reply:
Ah, I think I see.  Without this piece of the change, the new tests would fail because particular field types have binary data, and as-is we don't handle that correctly in SolrJ/EmbeddedSolrServer for wt=xml.

But figuring out how Solr should handle binary field data when using `wt=xml` is probably worth its own jira, for a few reasons:  (1) Other people in the community are likely to have opinions and want to chime in.  (2) It also, to be honest about my limitations, strays into areas of the code I don't know as well.  (3) It's fixing a different bug than these tests are meant for.

I'd like to see it get fixed, and I can work with you if you open a JIRA specifically for the binary-xml stuff.  I just don't think that fixing it here is the right approach.  Can you remove the binary-xml related changes, and either comment out the added tests that will fail, or add a TODO to add them once the underlying XML issue is fixed.  (If you file a separate JIRA and reference that in your TODO comment, others will have the context if they want it).

Reply:
> Ah, I think I see. Without this piece of the change, the new tests would fail because particular field types have binary data, and as-is we don't handle that correctly in SolrJ/EmbeddedSolrServer for wt=xml.
> 
Yes, of course!

> But figuring out how Solr should handle binary field data when using `wt=xml` is probably worth its own jira, for a few reasons: (1) Other people in the community are likely to have opinions and want to chime in. (2) It also, to be honest about my limitations, strays into areas of the code I don't know as well. (3) It's independent conceptually from the problem we started out trying to solve here (adding some atomic update tests to cover what we currently support).
> 
Added new issue SOLR-13762 and PR #883 

> I'd like to see it get fixed, and I can work with you if you open a JIRA specifically for the binary-xml stuff. I just don't think that fixing it here is the right approach. Can you remove the binary-xml related changes, and either comment out the added tests that will fail, or add a TODO to add them once the underlying XML issue is fixed. (If you file a separate JIRA and reference that in your TODO comment, others will have the context if they want it).

I was thinking about the same, so it was not much work anyway, and it is better separated now